src/main/java/lol/ui/Sound.java

```diff
@@ -0,0 +1,56 @@
+package lol.ui;
+
+
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.Clip;
+import javax.sound.sampled.FloatControl;
+import javax.sound.sampled.LineUnavailableException;
+import javax.sound.sampled.UnsupportedAudioFileException;
+import java.io.File;
+import java.io.IOException;
+
+//Source: Sound from Zapsplat.com
+//Mp3 is not supported
+public class Sound  {  
+   private File shootArrow;
+   private File swordHit;
+   private File explodeBuilding;
+   public Sound(){
+      this.shootArrow = new File("src\\main\\resources\\attack_sounds\\arrow.wav");
```

Comment:
Could you try : `new File("attack_sounds/arrow.wav")`? (path working on Linux and implicit resources path, like in `Sprites.java`.

Reply:
Yes I will test it now.

Reply:
It's throwing FileNotFoundException with "attack_sounds/arrow.wav" and "attack_sounds\\arrow.wav"

Reply:
Try with

```java
InputStream soundfile = ASCIIBattlefieldBuilder.class.getResourceAsStream(soundFileName);
if(soundfile == null) {
  throw new RuntimeException("The sound file `" + soundFileName + "` could not be found.");
}
```

And then:

```java
AudioInputStream inputStream = AudioSystem.getAudioInputStream(soundfile);
```

Reply:
I tried this method and no matter the path it always gave a null obj. I checked the getResourceAsStream and they use the URL class which I tried to use in the beginning and it always threw NullException, that's why I changed to the File class and the toURI.toURL(). Not sure why URL behaves like this.

Reply:
I think I fixed the bug, please try with:

```java
   public Sound(){
      this.shootArrow = openSoundFile("attack_sounds/arrow.wav");
      this.swordHit = openSoundFile("attack_sounds/sword.wav");
      this.explodeBuilding = openSoundFile("attack_sounds/explode.wav");
   }

   private File openSoundFile(String soundPath) {
      File soundfile = new File(getClass().getClassLoader().getResource(soundPath).getFile());
      if(soundfile == null) {
        throw new RuntimeException("The sound file `" + soundPath + "` could not be found.");
      }
      return soundfile;
   }
```

Reply:
Yes now it is working, thanks.